### ISSUE TITLE
[Bug]: DynamoDB replica point-in-time-recovery enable stopped working 

### ISSUE THREAD

**Issue by harmanbirdi** [hashicorp/terraform-provider-aws#28814]
### Terraform Core Version

1.3.6

### AWS Provider Version

4.40.0

### Affected Resource(s)

aws_dynamodb_table - for dynamodb-global-tables-2019

### Expected Behavior

PITR should get enabled on the replicas if `point_in_time_recovery` is set to `true`.
The plan reports it correctly, but it fails to set it. It works correctly in `4.39.0` version.
So, this is a regression.

### Actual Behavior

PITR failed to be enabled in "4.40.0" version onwards.

### Relevant Error/Panic Output Snippet

```shell
None
```

### Terraform Configuration Files

None

### Steps to Reproduce

- Use `"aws_dynamodb_table"` resource to create a table with replica without PITR enabled on the replica.
- Once the table has been created, switch replica point_in_time_recovery to true, and run apply again.
- You will see that it enables the PITR if you use 

```
aws = {
      source  = "hashicorp/aws"
      version = "4.39.0"
    }
```

but it will fail if the version is upped to `"4.40.0"`

### Debug Output

None

### Panic Output

None

### Would you like to implement a fix?

None

**Comment by YakDriver** [comment]
Related #26552
Related #24903

I was unable to reproduce this problem given the following configuration:

```
Terraform v1.3.7
+ provider registry.terraform.io/hashicorp/aws v4.51.0
```

```terraform
terraform {
  required_providers {
    aws = {
      source = "hashicorp/aws"
    }
  }
}

provider "aws" {
  region = "us-west-2"
}

provider "aws" {
  region = "us-east-1"
  alias  = "awsalternate"
}

provider "aws" {
  region = "us-east-2"
  alias  = "awsthird"
}

data "aws_region" "alternate" {
  provider = aws.awsalternate
}

data "aws_region" "third" {
  provider = aws.awsthird
}

resource "aws_dynamodb_table" "test" {
  name             = "bzrp"
  hash_key         = "TestTableHashKey"
  billing_mode     = "PAY_PER_REQUEST"
  stream_enabled   = true
  stream_view_type = "NEW_AND_OLD_IMAGES"

  attribute {
    name = "TestTableHashKey"
    type = "S"
  }

  point_in_time_recovery {
    enabled = true
  }

  replica {
    region_name            = data.aws_region.alternate.name
    point_in_time_recovery = true
  }

  replica {
    region_name            = data.aws_region.third.name
    point_in_time_recovery = true
  }
}
```

I tried setting `enabled = false`, `enabled = true`, changing it in the console and then re-applying, and many combinations. Always it worked correctly, setting the PITR enabled according to the value in the configuration for both the local table and replicas.

Let us know how your configuration differs from this.

Also, it is important to note that your Terraform AWS access credentials must be valid in all the regions where you make replicas. The AWS provider will make separate connections for each replica.

**Comment by harmanbirdi** [comment]
Thanks Dirk for sending this.  I shall try to distill my configuration that
exhibits the problem and report back.

**Comment by harmanbirdi** [comment]
I can confirm that it has been fixed in `v4.52.0` of `hashicorp/aws` provider module.
Thank you so much for fixing this.
